lib/jekyll/entry_filter.rb

```diff
@@ -99,13 +98,14 @@ def symlink_outside_site_source?(entry)
     # Check if an entry matches a specific pattern.
     # Returns true if path matches against any glob pattern, else false.
     def glob_include?(enumerator, entry)
-      entry_with_source = PathManager.join(site.source, entry)
+      current_directory = PathManager.join(site.source, base_directory)
```

Comment:
In the initializer, we duplicate given absolute path and strips `site.source`  from it  (scope for optimization).
And here, we're going to recreate the absolute path and cache it? Why not initialize a new instance variable containing the absolute path?

Reply:
Right! I'm tempted to change `derive_base_directory` to always be an absolute path, but it'll merit a deprecation warning and other changes (specially in `excluded?` and `relative_to_source`).  Do you think I should go that way or just add a new instance var?

Thanks for reviewing :)

Reply:
new instance variable to bypass all that deprecation. Something along the lines of `@absolute_base_dir`